Users/sogangul/bicep add multi repo solution - Publish Wiki Guidelines for Multi-repo Orchestration with CARML (#999)
Azure/ResourceModules: docs/wiki/ModulesUsage.md

    - [**Remote:** Azure CLI](#remote-azure-cli)
- [Orchestrate deployment](#orchestrate-deployment)
  - [Template-orchestration](#template-orchestration)

> - [Solution deployment](#solution-deployment)

Reviewer: Let's move under `Orchestrate deployment / Pipeline-orchestration / Multi-Repository-Approach`

Reviewer 2: Moved.

  ]
}
```

## Solution deployment

The bicep modules provided by this repo can also be clubbed together to create more complex infrastructures and in the form a completely reusable solutions or products.

### Sample solution

#### Summary

1. Below is an example which uses a _multi-repo_ approach
1. It fetches the _public_ **Azure/ResourceModules** repo for consuming bicep modules and uses the parameter files present in the _private_ **Contoso/MultiRepoTest** repo for deploying infrastructure
1. This example is for creating a network hub with following resources -
    1. First stage: **Deploy resource group**

>         1. Checkout 'Azure/ResourceModules' repo at root of the agent

Reviewer: ```suggestion
        1. Checkout 'Azure/ResourceModules' repo at root of the agents
```

          parameterFilePath: './MultiRepoTestParentFolder/network-hub-rg/Parameters/VirtualNetworkPeering/vnetb-to-vneta-peering.parameters.json'
          location: '${{ env.defaultLocation }}'
          resourceGroupName: '${{ env.resourceGroupName }}'
          subscriptionId: '${{ secrets.ARM_SUBSCRIPTION_ID }}'
          managementGroupId: '${{ secrets.ARM_MGMTGROUP_ID }}'
          removeDeployment: $(removeDeployment)
```

#### Notes

> 1. 'Azure/ResourceModules' repo has been checked out at the root location intentionally because the `deep-mm/set-variables@v1.0` task expects the _global.variables.json_ file in the _.github/variables/_ location. The GitHub Actions also expect the underlying utility scripts at a specific location
> 1. 'contoso/MultiRepoTest' repo has been checked out in a nested folder called as "MultiRepoTestParentFolder" to distinguish it from the folders from the other repo in the agent but can be downloaded at the root location too if desired
> 1. Comparison between IaCS and CARML -
>
>    Snippet of Resource Group deployment job from the _Solution_ repo of IaCS -
>

> >    ![IaCS_RGDeployJob](/docs/media/IaCS_DeployRGJob.png)

Reviewer 2: Removed.

>

> >    | Sr. no. | Topic| IaCS | CARML |

Reviewer: Remove table

Reviewer 2: Removed.

    }
}
```

> ## Solution deployment

Reviewer: ```suggestion
## Pipeline-orchestration
```

> The bicep modules provided by this repo can also be clubbed together to create more complex infrastructures and in the form a completely reusable solutions or products.

Reviewer: ```suggestion
The modules provided by this repo can be orchestrated to create more complex infrastructures and as such reusable solutions or products.
```

Reviewer: Overall, I', not sure if that fit's here. The section should be `Pipeline-orcehstration` and be a counter-part to the explaination what template-orchestration is (above).

Then, there should be a sub-header `# Multi-Repository approach` which explicitely explains this flavor

### Sample solution

#### Summary

> 1. Below is an example which uses a _multi-repository_ approach

Reviewer: ```suggestion
1. Below you can find an example which uses makes use of multiple repositories to orchestrat the deployment (also known as a _multi-repository_ approach)
```
